url.js

```diff
@@ -435,9 +435,9 @@
             (this.user && (
                 encode(this.user) + (this.pass && (':' + encode(this.pass))
             ) + '@')) +
-            (this.host && this.host) +
```

Comment:
The && operators usage here is not in the logical "and" operator sense.
"Both && and || result in the value of (exactly) one of their operands:
A && B returns the value A if A can be coerced into false; otherwise, it returns B.
A || B returns the value A if A can be coerced into true; otherwise, it returns B."
[Using logical operators with non boolean values](https://mariusschulz.com/blog/the-and-and-or-operators-in-javascript#using-logical-operators-with-non-boolean-values)
Thus code was correct. Please revert the commit.